src/app/src/components/DrawTools/EditToolbar.js

```diff
@@ -28,18 +28,22 @@ import {
     toggleEditMode,
     togglePolygonVisibility,
 } from '../../store/mapSlice.js';
+import { useDrawBoundary } from '../DrawContext.js';
 
 const POLYGON_BUTTON_WIDTH = 40;
 
 export default function EditToolbar() {
     const dispatch = useDispatch();
-    const { polygon, editMode, addPolygonMode } = useSelector(
+    const boundary = useDrawBoundary();
+    const { editMode, addPolygonMode, polygonIsVisible } = useSelector(
         state => state.map
     );
 
     const confirmDeleteDialogController = useDialogController();
     const editDialogController = useDialogController();
 
+    const hasShape = Boolean(boundary.submission?.shape);
```

Comment:
Consider using something more idiomatic of JavaScript rather than a TypeScript convention.

```suggestion
    const hasShape = !!boundary.submission?.shape;
```